_api-reference/security/configuration/update-configuration.md

```diff
@@ -0,0 +1,151 @@
+| `do_not_fail_on_forbidden` | Boolean | When `true`, returns empty results instead of a forbidden error settings the user is not authorized to access. Instead, failures are stored in the application logs. |
+| `do_not_fail_on_forbidden_empty` | Boolean | Similar to `do_not_fail_on_forbidden` but with specific behavior for empty results. |
+| `filtered_alias_mode` | String | Controls how document field filtering is applied on aliases. |
+| `hosts_resolver_mode` | String | Determines how hostname resolution is performed for security operations. |
+| `http` | Object | The HTTP-specific security configurations. |
+| `on_behalf_of` | Object | Configures a temporary access token for the duration of a user's session (advanced). |
+| `kibana` | Object | The configuration for OpenSearch Dashboards integration. |
+| `respect_request_indices_options` | Boolean | When `true`, respects index options specified in requests. |
+
+
+</details>
+
+
+## Example request
+
+The following example updates the security configuration to configure basic authentication and internal user database:
+
+```json
+PUT /_plugins/_security/api/securityconfig/config
+{
+  "dynamic": {
+    "filtered_alias_mode": "warn",
+    "disable_rest_auth": false,
+    "disable_intertransport_auth": false,
+    "respect_request_indices_options": false,
+    "opensearch-dashboards": {
+      "multitenancy_enabled": true,
+      "server_username": "kibanaserver",
+      "index": ".opensearch-dashboards"
+    },
+    "http": {
+      "anonymous_auth_enabled": false
+    },
+    "authc": {
+      "basic_internal_auth_domain": {
+        "http_enabled": true,
+        "transport_enabled": true,
+        "order": 0,
+        "http_authenticator": {
+          "challenge": true,
+          "type": "basic",
+          "config": {}
+        },
+        "authentication_backend": {
+          "type": "intern",
+          "config": {}
+        },
+        "description": "Authenticate via HTTP Basic against internal users database"
+      }
+    },
+    "auth_failure_listeners": {},
+    "do_not_fail_on_forbidden": false,
+    "multi_rolespan_enabled": true,
+    "hosts_resolver_mode": "ip-only",
+    "do_not_fail_on_forbidden_empty": false
+  }
+}
+```
+{% include copy-curl.html %}
+
+## Example response
+
+```json
+{
+  "status": "OK",
+  "message": "Configuration updated."
+}
+```
+
+## Response body fields
+
+The response body is a JSON object with the following fields:
+
+| Property | Data type | Description |
+| :--- | :--- | :--- |
+| `status` | String | Status of the request. A successful request returns "OK". |
+| `message` | String | A message describing the result of the operation. |
+
+## Usage notes
+
+The Update Configuration API provides direct access to modify the Security plugin's core configuration, but comes with significant risks:
+
+- **Prefer `securityadmin.sh`**: In most cases, you should use the `securityadmin.sh` script instead, which includes validation and safeguards.
```

Comment:
```suggestion
- **Prefer `securityadmin.sh`**: In most cases, you should use the `securityadmin.sh` script instead, which includes validations and safeguards to prevent misconfiguration.
```